lib/parser.dart

```diff
@@ -671,7 +683,9 @@ class _MappingTokenizer implements Iterator<String> {
       buff.write(_internal[i]);
     }
     buff.write('[31m');
-    buff.write(current ?? '');
+    try {
+      buff.write(current);
+    } on RangeError catch (_) {}
```

Comment:
Why can this start throwing?

Reply:
That is how I chose to handle invalid access to `current` when there are no more items.

We could alternatively choose some "default" value (the docs allow either behavior), but I think this is better unless we _have_ to go the other route.